src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -220,6 +221,7 @@ export class NotebookEditor implements INotebookEditor {
                 const v = await this.applicationShell.showInformationMessage(message, yes, no);
                 if (v === yes) {
                     this.restartingKernel = false;
+                    this.kernelInterruptedDontAskToRestart = true;
```

Comment:
I would reset this value after the next line, after line 225.
Or better, do it with a `try...finall...` & initialize in the `finally`.